### ISSUE TITLE
splunk-py-trace-bootstrap causing dependency issues (installs 0.28b0, wants 0.27b0)

### ISSUE THREAD

**Issue by NeilJed** [signalfx/splunk-otel-python#202]
Installing with `pip -v install splunk-opentelemetry[all]` installs version 1.40 and installs additional dependencys such as `opentelemetry-instrumentation==0.28b0`, etc.

However, when running `splunk-py-trace-bootstrap` after, it throws and error:

`RuntimeError: Dependency conflict found: opentelemetry-instrumentation-botocore 0.27b0 has requirement opentelemetry-instrumentation==0.27b0, but you have opentelemetry-instrumentation 0.28b0.
opentelemetry-instrumentation-botocore 0.27b0 has requirement opentelemetry-semantic-conventions==0.27b0, but you have opentelemetry-semantic-conventions 0.28b0.`

Trying to force install `opentelemetry-instrumentation-botocore 0.28b0` doesn't help as running the bootstrap script forcefully uninstalls it and tries to install 0.27b0 instead.

Only way around was to install splunk-opentelemetry version 1.30

**Comment by NeilJed** [comment]
I think I might have found the root cause of our issue. It seems to be that when we install our apps requirements ahead of the otel-collector the AWS `boto3` module is what is causing 0.28/0.27 conflict.

This simple Dockerfile will cause the error:

```
FROM python:3.10-slim


RUN pip install --upgrade pip wheel
RUN pip install --no-cache-dir boto3

ENV OTEL_SERVICE_NAME=my-service
RUN pip -v install splunk-opentelemetry[all]
RUN splunk-py-trace-bootstrap

CMD ["splunk-py-trace", "python", "./main.py" ]
```

Build output is:

```
...
#7 [4/5] RUN pip -v install splunk-opentelemetry[all]
#7 sha256:693ed0f1f1a579e4fde81ae7b655f0debc23add9d25adb887afaf87a227f5568
#7 0.290 Using pip 21.3.1 from /usr/local/lib/python3.10/site-packages/pip (python 3.10)
#7 0.382 Collecting splunk-opentelemetry[all]
#7 0.571   Downloading splunk_opentelemetry-1.4.0-py3-none-any.whl (29 kB)
#7 0.640 Collecting opentelemetry-instrumentation==0.28b0
#7 0.683   Downloading opentelemetry_instrumentation-0.28b0-py3-none-any.whl (22 kB)
#7 0.715 Collecting opentelemetry-semantic-conventions==0.28b0
#7 0.758   Downloading opentelemetry_semantic_conventions-0.28b0-py3-none-any.whl (23 kB)
...etc...

#8 [5/5] RUN splunk-py-trace-bootstrap
#8 sha256:b057fb022ca54e96be8cd4a5f384e751992cfc4267067785c444e068b732d6e9
#8 0.482 Collecting opentelemetry-instrumentation-aws-lambda==0.28b0
#8 0.673   Downloading opentelemetry_instrumentation_aws_lambda-0.28b0-py3-none-any.whl (10 kB)
#8 0.687 Requirement already satisfied: opentelemetry-semantic-conventions==0.28b0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-aws-lambda==0.28b0) (0.28b0)
#8 0.688 Requirement already satisfied: opentelemetry-instrumentation==0.28b0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-aws-lambda==0.28b0) (0.28b0)
#8 0.714 Collecting opentelemetry-propagator-aws-xray==1.0.1
#8 0.761   Downloading opentelemetry_propagator_aws_xray-1.0.1-py3-none-any.whl (10 kB)
#8 0.773 Requirement already satisfied: setuptools>=16.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.28b0->opentelemetry-instrumentation-aws-lambda==0.28b0) (57.5.0)
#8 0.774 Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.28b0->opentelemetry-instrumentation-aws-lambda==0.28b0) (1.13.3)
#8 0.774 Requirement already satisfied: opentelemetry-api~=1.4 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.28b0->opentelemetry-instrumentation-aws-lambda==0.28b0) (1.9.0)
#8 0.778 Requirement already satisfied: Deprecated>=1.2.6 in /usr/local/lib/python3.10/site-packages (from opentelemetry-api~=1.4->opentelemetry-instrumentation==0.28b0->opentelemetry-instrumentation-aws-lambda==0.28b0) (1.2.13)
#8 0.823 Installing collected packages: opentelemetry-propagator-aws-xray, opentelemetry-instrumentation-aws-lambda
#8 0.834 Successfully installed opentelemetry-instrumentation-aws-lambda-0.28b0 opentelemetry-propagator-aws-xray-1.0.1
...etc...

#8 3.436 Installing collected packages: opentelemetry-instrumentation-wsgi
#8 3.444 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#8 3.444 Successfully installed opentelemetry-instrumentation-wsgi-0.28b0
#8 3.676 Collecting opentelemetry-instrumentation-botocore==0.27b0
#8 3.859   Downloading opentelemetry_instrumentation_botocore-0.27b0-py3-none-any.whl (19 kB)
#8 3.906 Collecting opentelemetry-instrumentation==0.27b0
#8 3.947   Downloading opentelemetry_instrumentation-0.27b0-py3-none-any.whl (22 kB)
#8 3.984 Collecting opentelemetry-semantic-conventions==0.27b0
#8 4.025   Downloading opentelemetry_semantic_conventions-0.27b0-py3-none-any.whl (23 kB)
#8 4.035 Requirement already satisfied: opentelemetry-api~=1.3 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-botocore==0.27b0) (1.9.0)
#8 4.043 Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.27b0->opentelemetry-instrumentation-botocore==0.27b0) (1.13.3)
#8 4.043 Requirement already satisfied: setuptools>=16.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.27b0->opentelemetry-instrumentation-botocore==0.27b0) (57.5.0)
#8 4.045 Requirement already satisfied: Deprecated>=1.2.6 in /usr/local/lib/python3.10/site-packages (from opentelemetry-api~=1.3->opentelemetry-instrumentation-botocore==0.27b0) (1.2.13)
#8 4.088 Installing collected packages: opentelemetry-semantic-conventions, opentelemetry-instrumentation, opentelemetry-instrumentation-botocore
#8 4.088   Attempting uninstall: opentelemetry-semantic-conventions
#8 4.088     Found existing installation: opentelemetry-semantic-conventions 0.28b0
#8 4.089     Uninstalling opentelemetry-semantic-conventions-0.28b0:
#8 4.091       Successfully uninstalled opentelemetry-semantic-conventions-0.28b0
#8 4.098   Attempting uninstall: opentelemetry-instrumentation
#8 4.098     Found existing installation: opentelemetry-instrumentation 0.28b0
#8 4.100     Uninstalling opentelemetry-instrumentation-0.28b0:
#8 4.114       Successfully uninstalled opentelemetry-instrumentation-0.28b0
#8 4.132 ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
#8 4.132 splunk-opentelemetry 1.4.0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 splunk-opentelemetry 1.4.0 requires opentelemetry-semantic-conventions==0.28b0, but you have opentelemetry-semantic-conventions 0.27b0 which is incompatible.
#8 4.132 opentelemetry-sdk 1.9.0 requires opentelemetry-semantic-conventions==0.28b0, but you have opentelemetry-semantic-conventions 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-wsgi 0.28b0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-wsgi 0.28b0 requires opentelemetry-semantic-conventions==0.28b0, but you have opentelemetry-semantic-conventions 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-urllib 0.28b0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-urllib 0.28b0 requires opentelemetry-semantic-conventions==0.28b0, but you have opentelemetry-semantic-conventions 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-sqlite3 0.28b0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-logging 0.28b0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-dbapi 0.28b0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-dbapi 0.28b0 requires opentelemetry-semantic-conventions==0.28b0, but you have opentelemetry-semantic-conventions 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-aws-lambda 0.28b0 requires opentelemetry-instrumentation==0.28b0, but you have opentelemetry-instrumentation 0.27b0 which is incompatible.
#8 4.132 opentelemetry-instrumentation-aws-lambda 0.28b0 requires opentelemetry-semantic-conventions==0.28b0, but you have opentelemetry-semantic-conventions 0.27b0 which is incompatible.
#8 4.132 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#8 4.132 Successfully installed opentelemetry-instrumentation-0.27b0 opentelemetry-instrumentation-botocore-0.27b0 opentelemetry-semantic-conventions-0.27b0
#8 4.357 Collecting opentelemetry-instrumentation-grpc==0.28b0
#8 4.540   Downloading opentelemetry_instrumentation_grpc-0.28b0-py3-none-any.whl (19 kB)
#8 4.561 Requirement already satisfied: opentelemetry-sdk~=1.3 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-grpc==0.28b0) (1.9.0)
#8 4.562 Requirement already satisfied: opentelemetry-api~=1.3 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-grpc==0.28b0) (1.9.0)
#8 4.591 Collecting opentelemetry-instrumentation==0.28b0
#8 4.593   Using cached opentelemetry_instrumentation-0.28b0-py3-none-any.whl (22 kB)
#8 4.619 Collecting opentelemetry-semantic-conventions==0.28b0
#8 4.620   Using cached opentelemetry_semantic_conventions-0.28b0-py3-none-any.whl (23 kB)
#8 4.621 Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-grpc==0.28b0) (1.13.3)
#8 4.623 Requirement already satisfied: setuptools>=16.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.28b0->opentelemetry-instrumentation-grpc==0.28b0) (57.5.0)
#8 4.625 Requirement already satisfied: Deprecated>=1.2.6 in /usr/local/lib/python3.10/site-packages (from opentelemetry-api~=1.3->opentelemetry-instrumentation-grpc==0.28b0) (1.2.13)
#8 4.627 Requirement already satisfied: typing-extensions>=3.7.4 in /usr/local/lib/python3.10/site-packages (from opentelemetry-sdk~=1.3->opentelemetry-instrumentation-grpc==0.28b0) (4.0.1)
#8 4.671 Installing collected packages: opentelemetry-semantic-conventions, opentelemetry-instrumentation, opentelemetry-instrumentation-grpc
#8 4.671   Attempting uninstall: opentelemetry-semantic-conventions
#8 4.671     Found existing installation: opentelemetry-semantic-conventions 0.27b0
#8 4.672     Uninstalling opentelemetry-semantic-conventions-0.27b0:
#8 4.673       Successfully uninstalled opentelemetry-semantic-conventions-0.27b0
#8 4.680   Attempting uninstall: opentelemetry-instrumentation
#8 4.680     Found existing installation: opentelemetry-instrumentation 0.27b0
#8 4.682     Uninstalling opentelemetry-instrumentation-0.27b0:
#8 4.684       Successfully uninstalled opentelemetry-instrumentation-0.27b0
#8 4.702 Successfully installed opentelemetry-instrumentation-0.28b0 opentelemetry-instrumentation-grpc-0.28b0 opentelemetry-semantic-conventions-0.28b0
#8 4.702 ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
#8 4.702 opentelemetry-instrumentation-botocore 0.27b0 requires opentelemetry-instrumentation==0.27b0, but you have opentelemetry-instrumentation 0.28b0 which is incompatible.
#8 4.702 opentelemetry-instrumentation-botocore 0.27b0 requires opentelemetry-semantic-conventions==0.27b0, but you have opentelemetry-semantic-conventions 0.28b0 which is incompatible.
#8 4.702 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#8 4.921 Collecting opentelemetry-instrumentation-urllib3==0.28b0
#8 5.104   Downloading opentelemetry_instrumentation_urllib3-0.28b0-py3-none-any.whl (10 kB)
#8 5.121 Requirement already satisfied: opentelemetry-semantic-conventions==0.28b0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-urllib3==0.28b0) (0.28b0)
#8 5.121 Requirement already satisfied: opentelemetry-instrumentation==0.28b0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-urllib3==0.28b0) (0.28b0)
#8 5.122 Requirement already satisfied: wrapt<2.0.0,>=1.0.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-urllib3==0.28b0) (1.13.3)
#8 5.123 Requirement already satisfied: opentelemetry-api~=1.3 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-urllib3==0.28b0) (1.9.0)
#8 5.123 Requirement already satisfied: opentelemetry-util-http==0.28b0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation-urllib3==0.28b0) (0.28b0)
#8 5.127 Requirement already satisfied: setuptools>=16.0 in /usr/local/lib/python3.10/site-packages (from opentelemetry-instrumentation==0.28b0->opentelemetry-instrumentation-urllib3==0.28b0) (57.5.0)
#8 5.130 Requirement already satisfied: Deprecated>=1.2.6 in /usr/local/lib/python3.10/site-packages (from opentelemetry-api~=1.3->opentelemetry-instrumentation-urllib3==0.28b0) (1.2.13)
#8 5.191 Installing collected packages: opentelemetry-instrumentation-urllib3
#8 5.200 Successfully installed opentelemetry-instrumentation-urllib3-0.28b0
#8 5.200 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#8 5.421 Traceback (most recent call last):
#8 5.421   File "/usr/local/bin/splunk-py-trace-bootstrap", line 8, in <module>
#8 5.421     sys.exit(run())
#8 5.421   File "/usr/local/lib/python3.10/site-packages/splunk_otel/cmd/bootstrap.py", line 68, in run
#8 5.421     otel_run()
#8 5.421   File "/usr/local/lib/python3.10/site-packages/opentelemetry/instrumentation/bootstrap.py", line 154, in run
#8 5.421     cmd()
#8 5.421   File "/usr/local/lib/python3.10/site-packages/opentelemetry/instrumentation/bootstrap.py", line 123, in _run_install
#8 5.421     _pip_check()
#8 5.421   File "/usr/local/lib/python3.10/site-packages/opentelemetry/instrumentation/bootstrap.py", line 82, in _pip_check
#8 5.421     raise RuntimeError(f"Dependency conflict found: {pip_check}")
#8 5.421 RuntimeError: Dependency conflict found: opentelemetry-instrumentation-botocore 0.27b0 has requirement opentelemetry-instrumentation==0.27b0, but you have opentelemetry-instrumentation 0.28b0.
#8 5.421 opentelemetry-instrumentation-botocore 0.27b0 has requirement opentelemetry-semantic-conventions==0.27b0, but you have opentelemetry-semantic-conventions 0.28b0.
#8 5.421
```

If we don't install `boto3` before otel, everything works fine.

**Comment by owais** [comment]
Hi @NeilJed. Sorry for the inconvenience. I have released 1.9.1 upstream and 1.4.1 for splunk-otel-python which resolves this issue. Please let me know if it works now.

**Comment by NeilJed** [comment]
Thanks! Confirmed its fixed the issue for us.
